### PR TITLE
Added support for external URL token type

### LINKED ISSUE #2650: [APIView] Support for URL token type
Looking forward in the evolution of APIView, we would like to see a token type added for external links. Currently tokens can have a "navigateToId" property which controls jumping between things within the APIView, but we would like a token that displays URLs that will open links in a separate browser.

Some applications of this:
- Displaying the link to the source PR for generated APIViews as part of the APIView preamble (along with version etc.) The ultimate goal would actually be to link this to the source _Swagger_ PR, but an interim step would be linking to the language PR.
- Building off the previous, adding a (link) tag to certain lines that would provide a direct link to the source code line in the relevant PR.
- Building off that, potentially adding a (doc) tag to certain lines that would provide a direct link to the generated docs for a relevant PR.

Note that there's a whole lot more that would need to happen to fully implement these scenarios, but they underpinning all of them is the need for an external link mechanism.

### PR DESCRIPTION
**Pull request by Vailorium**
resolves #2650 

- Adds external link token start and end
  - Value for external link start is the link destination
  - Link opens in a new tab
- Added simple proof-of-concept with javadoc - converts all http/s links in javadoc to clickable links 

image

Important to note that you cannot nest <a> elements, so there could be issues with navigating between items, and linking to external prs as per #2650

### CODE REVIEW

**Review by JonathanGiles — APPROVED** [review]

File: src/dotnet/APIView/APIView/Model/CodeFileTokenKind.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ public enum CodeFileTokenKind
         TableColumnName = 24,
         TableCellBegin = 25,
         TableCellEnd = 26,
-        LeafSectionPlaceholder = 27
+        LeafSectionPlaceholder = 27,
+        ExternalLinkStart = 28,
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIView/Model/CodeFileTokenKind.cs:35]: I don't think we need two tokens here. Just one token type to show value is external link is good enough. Start and end are required when we need to include too many tokens wrapped within.

File: src/dotnet/APIView/APIView/CodeFileRenderer.cs [hunk 2]
```diff
@@ -215,6 +215,14 @@ private void Render(List<CodeLine> list, IEnumerable<CodeFileToken> node, bool e
                         leafSectionPlaceHolderNumber = (int)token.NumberOfLinesinLeafSection;
                         break;
 
+                    case CodeFileTokenKind.ExternalLinkStart:
+                        stringBuilder.Append($"<a target=\"_blank\" href=\"{token.Value}\">");
+                        break;
+
+                    case CodeFileTokenKind.ExternalLinkEnd:
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIView/CodeFileRenderer.cs:222]: No need of this token type. Please see below comment

**Review by praveenkuttappan — CHANGES_REQUESTED** [review]
See my comments about token kinds

File: src/dotnet/APIView/APIView/Model/CodeFileTokenKind.cs (hunk 1 above)
Comment (Vailorium) [reply src/dotnet/APIView/APIView/Model/CodeFileTokenKind.cs:35]: @praveenkuttappan the original issue stated that 

> Building off the previous, adding a (link) tag to certain lines that would provide a direct link to the source code line in the relevant PR.

Which I believe would require two tokens as any line could contain multiple tokens (correct me if I'm wrong).

I also believe that the separate tokens allows us to have links with a separate hyperlink and text (again, correct me if I'm wrong).

Happy to change it to one token though, let me know your thoughts.

Comment (JonathanGiles) [reply src/dotnet/APIView/APIView/Model/CodeFileTokenKind.cs:35]: I do think there is some value to the approach that @Vailorium has proposed in this PR. It means that we are not bound to show potentially long hyperlinks when that it the required hyperlink (nor do we have to manually shorten these long URLs so as to not invade our API reviews), and we can then instead shorten them to some link text, in the same way that web browsers enable this. In other words, it allows for us to separate the 'a href=..' from the link text itself.

**Review by praveenkuttappan — APPROVED** [review]

**Pull request merged**